mypy/treetransform.py

```diff
@@ -358,7 +358,12 @@ def copy_ref(self, new: RefExpr, original: RefExpr) -> None:
         new.fullname = original.fullname
         target = original.node
         if isinstance(target, Var):
-            target = self.visit_var(target)
+            # Do not transform references to global variables.
+            # TODO: is it really the best solution? Should we instead make
+            # a trivial copy, or do something completely different? See
+            # testGenericFunctionAliasExpand for an example where this is important.
+            if original.kind != GDEF:
+                target = self.visit_var(target)
```

Comment:
If you remove `visit_mypy_file` from the class (I think that it's unused), this will be completely reasonable, since this will be reference to something outside the AST node we are transforming.

Reply:
OK.

Reply:
Maybe also update the docstring that this can only be used to transform functions or classes, not top-level statements and such.